src/main/java/org/spongepowered/api/potion/PotionEffectType.java

```diff
@@ -25,10 +25,12 @@
 
 package org.spongepowered.api.potion;
 
+import org.spongepowered.api.attribute.AttributeSource;
+
 /**
  * Represents a possible type of {@link PotionEffect}.
  */
-public interface PotionEffectType {
+public interface PotionEffectType extends AttributeSource {
```

Comment:
@ST-DDT I understand that certain potion types have different attribute modifiers, but I hardly think that a PotionEffectType should be a source. If anything, that complicates finding out the attribute source. A Potion being used is an attribute modifier source, not the potion effect type.